hal/src/thumbv6m/sercom/v2/uart/capability.rs

```diff
@@ -0,0 +1,278 @@
+//! Capability
+//!
+//! Type-level constructs for specifiying the capabilities of a UART peripheral
+
+use crate::typelevel::Sealed;
+use bitflags::bitflags;
+use core::convert::TryFrom;
+
+/// Type-level enum representing the capabilities of a UART peripheral
+pub trait Capability: Sealed {
+    const FLAG_MASK: u8;
+    const STATUS_MASK: u16;
+}
```

Comment:
It looks like SPI, UART and I2C all have `u8` flag and `u16` status registers. Let's go ahead and elevate this to a `sercom` module trait.

Reply:
Wait, no. We can't make it general, because the corresponding implementers need custom constants. Grr